### DISCUSSION TITLE
Pickers search default value

### DISCUSSION THREAD

**Discussion by abdullaev388** [rsuite/rsuite#3104] [Ideas]
I am using SelectPicker with async search. When user searches, i send request to a server with search, then if user closes picker and opens it again, there will be empty search input, but results from server with search query. So i would like to have an opportunity to set default value of pickers search input.

**Comment by SevenOutman** [comment]
There's currently no way to populate the search input, but I think it's good to have one.

For now you can send request with empty search in `onClose` phase, so that the results are correct when user opens SelectPicker again.

**Reply by abdullaev388** [reply]
should i be waiting for you to provide some props for it?

**Reply by SevenOutman** [reply]
It should take quite some time because the existing props around searching (e.g. `onSearch`, `searchBy`) was designed based on the assumption that search keyword is handled inside SelectPicker. Need to find a way to allow outside control of search keyword while not breaking existing APIs.

**Comment by helveden** [comment]
Hello i search an example for use searchBy in `<CheckTreePicker />` ?

Thank You

**Reply by SevenOutman** [reply]
If I understand you correctly, here is an example of using `searchBy`

```jsx
const data = [
  { name: 'Adam', title: 'Front-end Developer' },
  { name: 'Bob', title: 'Project Manager' },
]

<CheckTreePicker
  data={data}
  // When user types in the search box
  // items with `.name` or `.title` matching the search word will be shown in search results
  searchBy={(keyword, _label, item) => {
    return item.name.includes(keyword) || item.title.includes(keyword);
  }}
/>
```

**Reply by helveden** [reply]
Thank you very much, some React JS operation is obscure for me, and I didn't understand the documentation as well as other stuff in the doc.

THANKS
